drivers/gpio/gpio_hpmicro.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2022 hpmicro
```

Comment:
Can this driver (and the pinctrl and uart drivers) get added directly into the main tree?

Reply:
Whoops - ignore this comment.  I see this commit is now adding to the main tree instead of a separate HAL.